python/ray/includes/metric.pxd

```diff
@@ -0,0 +1,45 @@
+from libcpp.string cimport string as c_string
+from libcpp.unordered_map cimport unordered_map
+from libcpp.vector cimport vector as c_vector
+
+cdef extern from "opencensus/tags/tag_key.h" nogil:
+    cdef cppclass CTagKey "opencensus::tags::TagKey":
+        @staticmethod
+        CTagKey Register(c_string &name)
+        const c_string &name() const
+
+cdef extern from "ray/stats/metric.h" nogil:
+    cdef cppclass CMetric "ray::stats::Metric":
+        CMetric(const c_string &name,
+               const c_string &description,
+               const c_string &unit,
+               const c_vector[CTagKey] &tag_keys)
+        c_string GetName() const
+        void Record(double value)
+        void Record(double value, 
+                unordered_map[c_string, c_string] &tags)
+
+    cdef cppclass CGauge "ray::stats::Gauge":
+        CGauge(const c_string &name,
```

Comment:
Just out of curiosity (Maybe I am not super familiar with syntax here). Won't there be any issue when we pass these values by reference? They seem to be GC'ed after Python function is terminated?

Reply:
> Just out of curiosity (Maybe I am not super familiar with syntax here). Won't there be any issue when we pass these values by reference? They seem to be GC'ed after Python function is terminated?

In order to keep consistency with cpp method, the CGauge class will be replaced in the generated cpp extension source file. So this is a cpp object not python.